lib/client.js

```diff
@@ -279,6 +280,25 @@
         };
     };
 
+    Client.prototype._onConnect = function () {
+
+        this._onDisconnectCalled = false;
+        this.onConnect();
+    };
+
+    Client.prototype._onDisconnect = function (log) {
+
+        if (!this._onDisconnectCalled) {       // guard against multiple invocation
+            this._onDisconnectCalled = true;
+            this.onDisconnect(this._willReconnect(), log);
```

Comment:
Shouldn't the first param here be the same as `log.willReconnect`, i.e. both need to be updated (or neither)?

My guess is that this initially was only passing a single `willReconnect` param and `log` was added later as a non-breaking change.